test/search_api/test_models.py

```diff
@@ -0,0 +1,745 @@
+    "visitId": "Test visitId",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "doi": "Test doi",
+    "startDate": "2000-12-31 00:00:00+00:00",
+    "title": "Test title",
+}
+
+INVESTIGATION_PARAMETER_ICAT_DATA = DATASET_PARAMETER_ICAT_DATA.copy()
+INVESTIGATION_PARAMETER_ICAT_DATA["stringValue"] = None
+INVESTIGATION_PARAMETER_ICAT_DATA["dateTimeValue"] = "2000-12-31 00:00:00+00:00"
+
+INVESTIGATION_TYPE_ICAT_DATA = {
+    "modId": "Test modId",
+    "description": "Test description",
+    "createId": "Test createId",
+    "name": "Test name",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "id": 1,
+    "modTime": "2000-12-31 00:00:00+00:00",
+}
+
+INVESTIGATION_USER_ICAT_DATA = {
+    "id": 1,
+    "modId": "Test modId",
+    "createId": "Test createId",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "role": "Test role",
+    "modTime": "2000-12-31 00:00:00+00:00",
+}
+
+KEYWORD_ICAT_DATA = {
+    "modId": "Test modId",
+    "createId": "Test createId",
+    "name": "Test name",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "id": 1,
+    "modTime": "2000-12-31 00:00:00+00:00",
+}
+
+PARAMETER_TYPE_ICAT_DATA = {
+    "pid": None,
+    "verified": True,
+    "unitsFullName": "Test unitsFullName",
+    "enforced": False,
+    "maximumNumericValue": 1.0,
+    "name": "Test name",
+    "modId": "Test modId",
+    "minimumNumericValue": 1.0,
+    "createId": "Test createId",
+    "applicableToDataCollection": False,
+    "applicableToInvestigation": False,
+    "applicableToDatafile": True,
+    "units": "Test units",
+    "id": 1,
+    "modTime": "2000-12-31 00:00:00+00:00",
+    "applicableToDataset": True,
+    "description": "Test description",
+    "valueType": "Test valueType",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "applicableToSample": False,
+}
+
+SAMPLE_ICAT_DATA = {
+    "pid": "None",
+    "modId": "Test modId",
+    "createId": "Test createId",
+    "name": "Test name",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "id": 1,
+    "modTime": "2000-12-31 00:00:00+00:00",
+}
+
+TECHNIQUE_ICAT_DATA = {
+    "name": "Test name",
+    "pid": "Test pid",
+    "description": "Test description",
+}
+
+USER_ICAT_DATA = {
+    "email": "Test email",
+    "name": "Test name",
+    "id": 1,
+    "modTime": "2000-12-31 00:00:00+00:00",
+    "familyName": None,
+    "modId": "Test modId",
+    "fullName": "Test fullName",
+    "orcidId": "Test orcidId",
+    "createId": "Test createId",
+    "createTime": "2000-12-31 00:00:00+00:00",
+    "affiliation": None,
+    "givenName": None,
+}
+
+
+AFFILIATION_PANOSC_DATA = {
+    "name": AFFILIATION_ICAT_DATA["name"],
+    "id": str(AFFILIATION_ICAT_DATA["id"]),
+    "address": AFFILIATION_ICAT_DATA["fullReference"],
+    "city": None,
+    "country": None,
+    "members": [],
+}
+
+DATASET_PANOSC_DATA = {
+    "pid": DATASET_ICAT_DATA["doi"],
+    "title": DATASET_ICAT_DATA["name"],
+    "creationDate": datetime.fromisoformat(DATASET_ICAT_DATA["createTime"]),
```

Comment:
Ah, I am using 3.8 so I didn't realise. Replacing all the usage of `datetime.fromisoformat` with `DateHandler.str_to_datetime_object()` works, thanks.

I keep forgetting to ask this but why are we still supporting 3.6? And 3.7 and 3.8? Python 3.10 recently got released so having to support all these version is too much I think and can get out of control at some point, especially with Python 3.11 on its way.

Reply:
So I believe we still need to support 3.6 because that's used in production and therefore we might as well run the CI on 3.7 and 3.8 too. I tried adding 3.10 to the CI back in October but came across various issues with dependencies (see https://github.com/icatproject-contrib/icat-ansible/issues/69). I think these are solvable by ignoring the `requirements.txt` and just installing Ansible ourselves (a newer version than stated in the requirements) but I haven't got round to it. Rolf has also kindly informed me that Payara 5 is required for Python 3.10 - while we run Payara 5 in production, I'm not sure if all facilities have made the move to Payara 5.